eclipse-collections-code-generator-maven-plugin/src/main/java/org/eclipse/collections/codegenerator/maven/GenerateMojo.java

```diff
@@ -31,12 +31,12 @@
     /**
      * Skips code generation if true.
      *
-     * @parameter expression="${skipCodeGen}"
+     * @parameter property="skipCodeGen"
      */
     private boolean skipCodeGen;
 
     /**
-     * @parameter expression="${project.build.directory}/generated-sources"
+     * @parameter default-value="${project.build.directory}/generated-sources"
```

Comment:
"default-value" is an expression that is used to calculate the value. "property" references a system property that can be overridden with 
```
mvn -DtemplateDirectory=someDirectory
```

I don't think it makes sense to be able to dynamically change the template directory from the command line. But you know best, I can add the property it if you want it. The build works either way.

Being able to skip the code generation from the command line with
```
mvn -DskipCodeGen=true
```
on the other hand makes sense to me.